src/test/ReactTestUtils.js

```diff
@@ -178,9 +178,6 @@ var ReactTestUtils = {
    * @return {array} an array of all the matches.
    */
   scryRenderedDOMComponentsWithClass: function(root, classNames) {
-    if (!Array.isArray(classNames)) {
```

Comment:
@sarbbottam I also thought about doing it that way. The reason that I did it the way I did was to make sure the errors were as useful and specific as possible. For example. If we were to do it the way that you're recommending, if someone makes a mistake and entered the className `foo` as the only argument. They will no longer receive the helpful error that states that `foo` is not an instance of a component but will be presented with this new error instead `Invariant Violation: TestUtils.scryRenderedDOMComponentsWithClass expects a className a second argument` , which, to me is more ambiguous and unclear of what part of their test they need to fix. I think the second (new) error is more helpful when it is only presented to the user after we have verified that the first argument is a valid instance of a component. That's my reasoning but I'm opening to updating the code to your implementation if that's the consensus. Thanks for the feedback!